sim/dex.ts

```diff
@@ -314,6 +314,24 @@ class ModdedDex {
 		}
 	}
 
+	getEffectivenessCode(move: Move, target: Pokemon): number {
+		if (!move.ignoreImmunity || (move.ignoreImmunity !== true && !move.ignoreImmunity[move.type])) {
+			if (!this.getImmunity(move, target)) {
+				return 3; // No effect
+			}
+		}
+		if (move.category === 'Status' || move.ohko || move.damageCallback || move.damage) {
+			return 0; // Effective
+		}
+		const baseMod = this.getEffectiveness(move, target);
+		// @ts-ignore - ModdedDex stubs Battle
+		const moveMod = move.onEffectiveness && move.onEffectiveness.call(this, baseMod, null, move.type, move);
+		const finalTypeMod = typeof moveMod === 'number' ? moveMod : baseMod;
+		if (finalTypeMod > 0) return 1; // Super effective
```

Comment:
Sending numeric enums over the protocol seems like a bad idea. At least send a string enum.